korangar/src/inventory/skills.rs

```diff
@@ -1,17 +1,91 @@
 use std::sync::Arc;
 
+use hashbrown::HashMap;
 use korangar_interface::element::StateElement;
 use ragnarok_packets::{ClientTick, SkillId, SkillInformation, SkillLevel, SkillType};
-use rust_state::RustState;
+use rust_state::{Path, RustState, Selector};
 
 use crate::loaders::{ActionLoader, Sprite, SpriteLoader};
+use crate::state::ClientState;
 use crate::world::{Actions, SpriteAnimationState};
 
+pub struct LearnedSkillPath<A, B> {
+    learnable_skill_path: A,
+    skills_path: B,
+}
+
+impl<A, B> LearnedSkillPath<A, B> {
+    pub fn new(learnable_skill_path: A, skills_path: B) -> Self {
+        Self {
+            learnable_skill_path,
+            skills_path,
+        }
+    }
+}
+
+impl<A, B> Clone for LearnedSkillPath<A, B>
+where
+    A: Clone,
+    B: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            learnable_skill_path: self.learnable_skill_path.clone(),
+            skills_path: self.skills_path.clone(),
+        }
+    }
+}
+
+impl<A, B> Copy for LearnedSkillPath<A, B>
+where
+    A: Copy,
+    B: Copy,
+{
+}
+
+impl<A, B> Path<ClientState, LearnedSkill, false> for LearnedSkillPath<A, B>
+where
+    A: Path<ClientState, LearnableSkill, false>,
+    B: Path<ClientState, Vec<LearnedSkill>>,
+{
+    fn follow<'a>(&self, state: &'a ClientState) -> Option<&'a LearnedSkill> {
+        let learnable_skill = self.learnable_skill_path.follow(state)?;
+        let learnable_skill_id = learnable_skill.skill_id;
+
+        // SAFETY:
+        // Unwrap is safe here because of the bounds.
```

Comment:
Seeing how this is repeated 4 times, maybe an inlined utility would be a better place to store it? (if not too costly)